arrow-core/src/main/kotlin/arrow/core/extensions/list/monad/ListKMonad.kt

```diff
@@ -0,0 +1,282 @@
+}
+
+@JvmName("mproduct")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("flatMap { a -> arg1(a).map { b -> Tuple2(a, b) } }"))
+fun <A, B> List<A>.mproduct(arg1: Function1<A, Kind<ForListK, B>>): List<Tuple2<A, B>> =
+    arrow.core.extensions.list.monad.List.monad().run {
+  arrow.core.ListK(this@mproduct).mproduct<A, B>(arg1) as
+    kotlin.collections.List<arrow.core.Tuple2<A, B>>
+}
+
+@JvmName("ifM")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("ifM(arg1, arg2)", "arrow.core.ifM"))
+fun <B> List<Boolean>.ifM(arg1: Function0<Kind<ForListK, B>>, arg2: Function0<Kind<ForListK, B>>):
+    List<B> = arrow.core.extensions.list.monad.List.monad().run {
+  arrow.core.ListK(this@ifM).ifM<B>(arg1, arg2) as kotlin.collections.List<B>
+}
+
+@JvmName("selectM")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("selectM(arg1)", "arrow.core.selectM"))
+fun <A, B> List<Either<A, B>>.selectM(arg1: List<Function1<A, B>>): List<B> =
+    arrow.core.extensions.list.monad.List.monad().run {
+  arrow.core.ListK(this@selectM).selectM<A, B>(arrow.core.ListK(arg1)) as kotlin.collections.List<B>
+}
+
+@JvmName("select")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("select(arg1)", "arrow.core.select"))
+fun <A, B> List<Either<A, B>>.select(arg1: List<Function1<A, B>>): List<B> =
+    arrow.core.extensions.list.monad.List.monad().run {
+  arrow.core.ListK(this@select).select<A, B>(arrow.core.ListK(arg1)) as kotlin.collections.List<B>
+}
+
+/**
+ * cached extension
+ */
+@PublishedApi()
+internal val monad_singleton: ListKMonad = object : arrow.core.extensions.ListKMonad {}
+
+object List {
+  /**
+   *  ank_macro_hierarchy(arrow.typeclasses.Monad)
+   *
+   *  [Monad] abstract over the ability to declare sequential computations that are dependent in the order or
+   *  the results of previous computations.
+   *
+   *  Given a type constructor [F] with a value of [A] we can compose multiple operations of type
+   *  `Kind<F, ?>` where `?` denotes a value being transformed.
+   *
+   *  This is true for all type constructors that can support the [Monad] type class including and not limited to
+   *  [IO], [ObservableK], [Option], [Either], [List] ...
+   *
+   *  [The Monad Tutorial](https://arrow-kt.io/docs/patterns/monads/)
+   */
+  @Suppress(
+    "UNCHECKED_CAST",
+    "NOTHING_TO_INLINE"
+  )
+  @Deprecated("Monad typeclasses is deprecated. Use concrete methods on List")
```

Comment:
```suggestion
  @Deprecated("Monad typeclasses are deprecated. Use concrete methods on List")
```
Or maybe use a constant message for all the same since all kinded type classes are deprecated.